catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/DownloadUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog.hive;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.util.Optional;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+public class DownloadUtils {
+
+  private DownloadUtils() {}
+
+  public static void downloadFile(
+      String keyTabUri, File keyTabFile, int timeout, Configuration conf) throws IOException {
```

Comment:
Why the parameter should be keyTabXXX, it is a common method. Please carefully review your code again.

Also, can you please move this method to common module. It actually can be used by other modules.

Reply:
We can't put this to common module. If we put this to common module, the Hadoop class will be loaded by the root JVM. We can't isolate the class of Hadoop.